src/main/java/uk/nhs/ctp/service/ReferralRequestService.java

```diff
@@ -1,58 +1,57 @@
 import uk.nhs.ctp.service.fhir.ReferenceService;
-import uk.nhs.ctp.transform.ReferralRequestEntityTransformer;
-import uk.nhs.ctp.utils.ErrorHandlingUtils;
+import uk.nhs.ctp.service.fhir.StorageService;
+import uk.nhs.ctp.utils.RetryUtils;
 
 @Service
 @AllArgsConstructor
+@Slf4j
 public class ReferralRequestService {
 
-  private ReferralRequestEntityTransformer referralRequestEntityTransformer;
-  private ReferralRequestRepository referralRequestRepository;
-  private FhirContext fhirContext;
+  private AppointmentService appointmentService;
+  private StorageService storageService;
   private ReferenceService referenceService;
 
-  public ReferralRequest makeAbsolute(ReferralRequest referralRequest) {
-    IdType idElement = referralRequest.getIdElement();
-    Preconditions.checkArgument(idElement.isAbsolute(), "Referral request must have absolute ID");
-    String baseUrl = idElement.getBaseUrl();
-
-    ReferralRequest copy = referralRequest.copy();
-    copy.setIdElement(null);
-    referenceService.resolveRelative(baseUrl, copy);
-
-    return copy;
+  public void updateServiceRequested(SelectedServiceRequestDTO requestDTO) {
+    log.info("Setting selected HealthcareService for case " + requestDTO.getCaseId());
+    ReferralRequest referralRequest = getByCaseId(requestDTO.getCaseId()).orElseThrow();
+    List<CodeableConcept> serviceRequested = requestDTO.getServiceTypes().stream()
+        .map(codeDTO ->
+            new CodeableConcept().addCoding(
+                new Coding(codeDTO.getSystem(), codeDTO.getCode(), codeDTO.getDisplay())))
+        .collect(Collectors.toList());
+    referralRequest.setServiceRequested(serviceRequested);
+    referralRequest.setRecipient(singletonList(new Reference(requestDTO.getSelectedServiceId())));
+    appointmentService.create(referralRequest); //Create a static appointment for the referral request.
+    storageService.updateExternal(referralRequest);
   }
 
-  /**
-   * Applies an update to the embedded ReferralRequest resource by deserializing it, applying the
-   * update function and then re-serializing it.
-   */
-  public void update(ReferralRequestEntity entity, Consumer<ReferralRequest> updateFn) {
-    ReferralRequest referralRequest = referralRequestEntityTransformer.transform(entity);
-    updateFn.accept(referralRequest);
-    IParser fhirParser = fhirContext.newJsonParser();
-    entity.setResource(fhirParser.encodeResourceToString(referralRequest));
-  }
-
-  @Transactional
-  public ReferralRequest get(Long id) {
-    ReferralRequestEntity referralRequestEntity =
-        referralRequestRepository.findOne(id);
-    ErrorHandlingUtils.checkEntityExists(referralRequestEntity, "ReferralRequest");
-
-    return referralRequestEntityTransformer.transform(referralRequestEntity);
+  public Optional<ReferralRequest> getByCaseId(Long id) {
+    return RetryUtils.retry(() -> storageService.getClient().search()
+        .forResource(ReferralRequest.class)
```

Comment:
If this returns all RRs, including cancelled ones, I think it needs to filter to only 'active' status. Are we certain there will only ever be a single RR?

Reply:
previously it was a one-one mapping so would only ever return one. I think now it'll actually return the interim ones as well. Depends on what we want to include in the ER (we have no requirements) but my suggestion would actually just to be to add them all.